### PR TITLE
Add Post API tests

### PR DESCRIPTION
**Pull request by Uptaker**

### CODE REVIEW

File: backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java
Comment (mariliisaar) [inline backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java:67]: Just curious - what does this line do?

File: backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java
Comment (mariliisaar) [inline backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java:47]: Feeling like a bit of a broken record here but should we, perhaps, test a list with at least two posts?

File: backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java
Comment (mariliisaar) [inline backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java:99]: Maybe this is where we should check that length is definitely 1? And we're not accidentally getting back multiple items? Might be overkill, of course 😄

File: backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java
Comment (mariliisaar) [inline backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java:121]: If I get this right, this is a new post, in addition to p1? If that's the case, then I'd change the text of this new post so the two do not match and the test below actually differentiates between them 🙂

File: backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java
Comment (mariliisaar) [inline backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java:160]: I think I'm getting confused here but I don't see us actually changing anything about the post?

File: backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java
Comment (mariliisaar) [inline backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java:212]: Shouldn't the parameter at the end here be username or something?

**Review by mariliisaar — COMMENTED** [review]
I had a couple of questions, hope you won't mind explaining where I'm being dumb before I approve 😄

File: backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java
Comment (Uptaker) [reply backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java:67]: Prints the result, eg body, status for debugging

File: backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java
Comment (Uptaker) [reply backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java:99]: It returns an individual post, not a list of posts, there's no size here :)

File: backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java
Comment (Uptaker) [reply backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java:212]: That's true - the fallacy of copy/paste :P

File: backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java
Comment (Uptaker) [reply backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java:121]: P1 is the post that will be returned - not yet created in the context of the repository. We construct the output of the service layer and then go to the endpoint of the controller, which then calls that service and returns the constructed output (eg a new post, p1) and the controller has to get the same output.

Another way is to test with the H2 database and also mock the repository, but that has its own plethora of complexity!

File: backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java
Comment (Uptaker) [reply backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java:47]: Sure, but that'll have to wait until the next week as I'm not available - unless you'd like to do it!

File: backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java
Comment (Uptaker) [reply backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java:160]: We mock the actual output of the service layer - there's no repository or database in tests that would change it. This makes sure that the request was valid and actually outputs the above post, even if nothing was changed. I can indeed change the title and add a line that changes p1's title, but it would accomplish the same thing, as it's still a valid request.

I can add a todo if it's necessary?

File: backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java
Comment (mariliisaar) [reply backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java:47]: It was just something to note for the future :)

File: backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java
Comment (mariliisaar) [reply backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java:99]: Fair enough

File: backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java
Comment (mariliisaar) [reply backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java:121]: Ah OK, I'd probably mock out both states (original and then edited) and check that it actually changes but that's probably just me.

File: backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java
Comment (mariliisaar) [reply backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java:160]: Ah OK, I'd probably mock out both states (original and then edited) and check that it actually changes but that's probably just me.
I shall leave it up to you to decide whether you can be bothered or not. I don't think it's a deal breaker.

File: backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java
Comment (mariliisaar) [reply backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java:121]: Ohhh, I misunderstood the code then. My bad.

File: backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java
Comment (mariliisaar) [reply backend/src/test/java/ee/tlu/forum/controller/PostControllerTest.java:212]: I'll approve but perhaps change this little bit when you next get to the Posts controller 😄

**Review by mariliisaar — APPROVED** [review]

**Pull request merged**
